app/src/main/java/org/dhis2/usescases/login/LoginPresenter.kt

```diff
@@ -253,15 +252,16 @@ class LoginPresenter(
         }
     }
 
-    private fun handleError(throwable: Throwable) {
+    private fun handleError(
+        throwable: Throwable,
+        serverUrl: String,
+        userName: String,
+        pass: String
+    ) {
         Timber.e(throwable)
         if (throwable is D2Error && throwable.errorCode() == D2ErrorCode.ALREADY_AUTHENTICATED) {
-            preferenceProvider.apply {
-                setValue(SESSION_LOCKED, false)
-
-                setValue(PIN, null)
-            }
-            view.alreadyAuthenticated()
+            userManager?.d2?.userModule()?.blockingLogOut()
+            logIn(serverUrl, userName, pass)
```

Comment:
do you think we can unit test this ? :+1:

Reply:
It is not that easy as the test goes into a loop... Could be done with too many changes in the presenter code for a fix.
@mmmateos , @JaimeToca , @ferdyrod Maybe you can have a try?